airflow-core/src/airflow/api_fastapi/core_api/datamodels/plugins.py

```diff
@@ -69,20 +69,35 @@ class AppBuilderMenuItemResponse(BaseModel):
     category: str | None = None
 
 
-class ExternalViewResponse(BaseModel):
-    """Serializer for IFrame Plugin responses."""
+class BaseUIResponse(BaseModel):
+    """Base serializer for UI Plugin responses."""
 
     model_config = ConfigDict(extra="allow")
 
     name: str
-    href: str
     icon: str | None = None
     icon_dark_mode: str | None = None
     url_route: str | None = None
     category: str | None = None
     destination: Literal["nav", "dag", "dag_run", "task", "task_instance"] = "nav"
 
 
+class ExternalViewResponse(BaseUIResponse):
+    """Serializer for External View Plugin responses."""
+
+    model_config = ConfigDict(extra="allow")
```

Comment:
Why actually do we allow extras here?

Reply:
This is how plugin attributes are built, every members of the Plugin allow extra attributes and are serialized and returned, so newly added attributes (react_apps, fastapi_apps, eternal_views) also follow the same pattern. 

It was here for legacy so I kept the behavior, meaning that you can attach some extra metadata to your plugin members in case you need it.